src/main/java/org/sonatype/nexus/ci/nxrm/v3/MoveComponentsStep.java

```diff
@@ -0,0 +1,138 @@
+import com.sonatype.nexus.api.repository.v3.RepositoryManagerV3Client;
+
+import org.sonatype.nexus.ci.config.NxrmVersion;
+import org.sonatype.nexus.ci.util.FormUtil;
+import org.sonatype.nexus.ci.util.NxrmUtil;
+
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.AbstractProject;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.tasks.BuildStepDescriptor;
+import hudson.tasks.Builder;
+import hudson.util.FormValidation;
+import hudson.util.ListBoxModel;
+import jenkins.tasks.SimpleBuildStep;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+
+import static com.sonatype.nexus.api.common.ArgumentUtils.checkArgument;
+import static com.sonatype.nexus.api.common.NexusStringUtils.isNotBlank;
+import static hudson.model.Result.FAILURE;
+import static java.util.stream.Collectors.joining;
+import static org.sonatype.nexus.ci.nxrm.Messages.Common_Validation_NexusInstanceIDRequired;
+import static org.sonatype.nexus.ci.nxrm.Messages.Common_Validation_Staging_TagNameRequired;
+import static org.sonatype.nexus.ci.nxrm.Messages.MoveComponents_DisplayName;
+import static org.sonatype.nexus.ci.nxrm.Messages.MoveComponents_Validation_DestinationRequired;
+import static org.sonatype.nexus.ci.util.RepositoryManagerClientUtil.nexus3Client;
+
+public class MoveComponentsStep
+    extends Builder
+    implements SimpleBuildStep
+{
+  private final String nexusInstanceId;
+
+  private final String tagName;
+
+  private final String destination;
+
+  @DataBoundConstructor
+  public MoveComponentsStep(final String nexusInstanceId,  final String tagName, final String destination)
+  {
+    this.nexusInstanceId = checkArgument(nexusInstanceId, isNotBlank(nexusInstanceId),
+        Common_Validation_NexusInstanceIDRequired());
+    this.destination = checkArgument(destination, isNotBlank(destination),
+        MoveComponents_Validation_DestinationRequired());
+    this.tagName = checkArgument(tagName, isNotBlank(tagName), Common_Validation_Staging_TagNameRequired());
+  }
+
+  public String getNexusInstanceId() {
+    return nexusInstanceId;
+  }
+
+  public String getTagName() {
+    return tagName;
+  }
+
+  public String getDestination() {
+    return destination;
+  }
+
+  @Override
+  public void perform(@Nonnull final Run run, @Nonnull final FilePath workspace, @Nonnull final Launcher launcher,
+                      @Nonnull final TaskListener listener) throws InterruptedException, IOException
+  {
+    try {
+      RepositoryManagerV3Client client = nexus3Client(nexusInstanceId);
+      List<ComponentInfo> components = client.move(destination, tagName);
+      listener.getLogger().println("Move successful. Destination: '" + destination + "' Components moved:\n" +
+          components.stream()
+          .map(c -> c.getGroup() + ":" + c.getName() + ":" + c.getVersion())
+          .collect(joining("\n")));
+    }
```

Comment:
Question: I added logging output to dump the list of components that were impacted by the move to be consistent with how the rest endpoint behaves if calling it directly. If we think this should be axed for formatted differently let me know.

Reply:
I like this, or at the very least having some sort of success log message. If you wanted to minimize at all you could reduce it to the destination and number of components moved.

Reply:
I'm going to leave as is unless someone feels strongly to reduce it. Personally I would appreciate having the list of moved components.